stdlib/public/Concurrency/AsyncStreamBuffer.swift

```diff
@@ -123,7 +123,8 @@ extension AsyncStream {
       lock()
       let limit = state.limit
       let count = state.pending.count
-      if let continuation = state.continuation {
+      
+      if let continuation = state.continuations.first {
```

Comment:
What if this were:

```swift
if !state.continuations.isEmpty {
  let continuation = state.continuations.removeFirst()
  ...
```

Then you'd avoid accidentally having some branch that fails to pop the continuation.